src/neo/Network/P2P/RemoteNode.ProtocolHandler.cs

```diff
@@ -287,18 +287,20 @@ private void OnGetHeadersMessageReceived(GetBlockByIndexPayload payload)
         private void OnInventoryReceived(IInventory inventory)
         {
             pendingKnownHashes.Remove(inventory.Hash);
-            knownHashes.Add(inventory.Hash);
-            system.TaskManager.Tell(inventory);
-            system.Blockchain.Tell(inventory, ActorRefs.NoSender);
             switch (inventory)
             {
                 case Transaction transaction:
+                    knownHashes.Add(inventory.Hash);
```

Comment:
Why move this line to here?

Reply:
Because we can't add the block to known hashes, maybe we need to ask for it later

Reply:
Move to L302?